psh/test-runfile.py

```diff
@@ -0,0 +1,134 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh runfile applet test
+#
+# Copyright 2022 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import pexpect
+import psh.tools.psh as psh
+import trunner.config as config
+from psh.tools.psh import EOT
+from psh.tools.common import CHARS, get_rand_strings
+
+
+def get_root_dirs(p):
+    root_dirs = []
+    root_content = psh.ls(p)
+    for root_file in root_content:
+        if root_file.is_dir:
+            root_dirs.append(root_file.name)
+    return root_dirs
+
+
+def assert_nonexistent(p, root_dirs):
+    cmd = '/nonexistent_file'
+    msg = 'Wrong error message, when executing nonexistent file'
+    psh.assert_cmd(p, cmd, f'psh: {cmd} not found', msg)
+
+    rfnames = get_rand_strings(CHARS, 10)
+    for rfname in rfnames:
+        cmd = '/nonexistent_file'
+        msg = f'Prompt not seen after executing: /{rfname}'
+        psh.assert_prompt_after_cmd(p, f'/{rfname}', msg)
+        psh.assert_cmd_failed(p)
+
+        msg = f'Wrong error message, when executing: /{rfname}'
+        psh.assert_cmd(p, f'/{rfname}', f'psh: /{rfname} not found', msg)
+        for root_dir in root_dirs:
+            msg = f'Prompt not seen after executing: /{root_dir}/{rfname}'
+            psh.assert_prompt_after_cmd(p, f'/{rfname}', msg)
+            psh.assert_cmd_failed(p)
+
+
+def assert_dirs(p, root_dirs):
+    dirname = 'test_runfile_dir'
+    msg = f'Creating the following directory failed: {dirname}'
+    psh.assert_cmd(p, f'mkdir {dirname}', '', msg)
+    psh.assert_cmd_successed(p)
+
+    msg = f'Prompt not seen after executing: /{dirname}'
+    psh.assert_prompt_after_cmd(p, f'/{dirname}', msg)
+    psh.assert_cmd_failed(p)
+
+    # psh.assert_cmd(p, f'/{dirname}', f'psh: /{dirname} is not an executable')
+
+    for root_dir in root_dirs:
+        msg = f'Prompt not seen after executing: /{root_dir}'
+        psh.assert_prompt_after_cmd(p, f'/{dirname}', msg)
+        psh.assert_cmd_failed(p)
```

Comment:
Extend psh.assert_* commands to limit code repetition. This code is easier to read and write:

```suggestion
    psh.assert_cmd(p, "mkdir test_runfile_dir", result='success')
    psh.assert_cmd(p, "/dirname", result='fail')
    
    for root_dir in root_dirs:
        psh.assert_cmd(p, "/dirname", result='fail')
```

Reply:
I would like to leave both assert_cmd() and assert_prompt_after_cmd(), because in `assert_cmd()` it's crucial to catch everything between cmd and prompt, when assert_prompt_after_cmd() does not care about this output between. I think adding `result` argument for both of them would be a good idea.